tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -1332,17 +1368,19 @@ private[optimizer] abstract class OptimizerCore(
         true
 
       case _ =>
-        arg.tpe.base match {
-          case ClassType("s_Predef$$less$colon$less" | "s_Predef$$eq$colon$eq") =>
-            true
-          case _ =>
-            false
-        }
+        isTypeLikelyOptimizable(arg.tpe)
+    }
+
+    receiverAndArgs.exists(isLikelyOptimizable) || {
+      target.toString == "s_reflect_ClassTag$.apply__jl_Class__s_reflect_ClassTag" &&
+      (receiverAndArgs.tail.head match {
+        case PreTransTree(ClassOf(_), _) => true
+        case _                           => false
+      })
     }
-    receiverAndArgs.exists(isLikelyOptimizable)
   }
 
-  private def inline(allocationSite: Option[AllocationSite],
+  private def inline(allocationSites: Set[AllocationSite],
```

Comment:
Yes, possibly, but then the possible number of different `allocationsSites` grows *much more* compared to the size of the program. Here, already, we have gone from N to 2^N where N is the number of `new`s in the program. With a List, this grows to M^N where M is the maximum number of parameters in a function.

As long as I don't encounter an actual use case where `List` would improve on the resulting code, it doesn't seem worth risking this explosion for nothing. Hence the `Set`.